src/main/java/com/gachigang/ontherun/service/BusinessService.java

```diff
@@ -32,24 +33,23 @@ public List<Business> getAllBusiness(PageRequest pageRequest) {
     @Transactional(readOnly = true)
     public Business getBusinessById(@NonNull final Long id) {
         return businessRepository.findById(id)
-                .orElseThrow(() -> new IllegalArgumentException("Business with id " + id + " not found"));
+                .orElseThrow(() -> new NotFoundException(Business.class));
     }
 
     @Transactional
-    public Business updateBusiness(UpdateBusinessRequest businessRequest, @NonNull Long id) {
-        Business businessById = getBusinessById(id);
-        businessById.setName(businessRequest.getName());
-        businessById.setCountry(businessRequest.getCountry());
-        businessById.setCity(businessRequest.getCity());
-        if (!businessRequest.getOwners().equals(businessById.getOwners().stream().map(User::getId).collect(Collectors.toSet()))) {
-            businessById.getOwners().addAll(userRepository.findAllByIdIn(businessRequest.getOwners()));
-        }
-        return businessRepository.save(businessById);
+    public Business updateBusiness(@NonNull final BusinessDto businessDto,
+                                   @NonNull final Long id) {
+        Business business = getBusinessById(id);
+
+        businessMapper.updateBusiness(business, businessDto);
```

Comment:
Seems like you forgot to update owners here

Reply:
I added a field where the method will set the owners.